xdsl/transforms/reconcile_unrealized_casts.py

```diff
@@ -89,27 +94,32 @@ def gen_all_uses_cast(
         rewriter.erase_op(cast)
 
 
+@dataclass
 class ReconcileUnrealizedCastsPattern(RewritePattern):
     """
     Removes the chains of `builtin.unrealized_conversion_cast` operations
     that are no longer necessary and that start with the matched
     `builtin.unrealized_conversion_cast`.
     """
 
+    with_warnings: bool = False
```

Comment:
Similarly to below, this is the way dataclasses specifies that the properties must be passed as keywords to the dataclass generated init. The alternative is `with_warnings: bool = field(default=True, kw_only=True)`, matter of style
```suggestion
    _: KW_ONLY
    with_warnings: bool = False
```